cmake/modules/extensions.cmake

```diff
@@ -4298,3 +4264,16 @@ macro(zephyr_check_flags_exclusive function prefix)
       )
   endif()
 endmacro()
+
+# Internal helper macro for zephyr_linker*() functions.
+# The macro will append a ${list} of "argument;value" pairs to the linker TARGET
+# property named as ${linker_target_property}.
+# The var names within the list are passed as ARGN.
+macro(zephyr_linker_add_property list linker_target_property)
+  set(${list})
+  list(APPEND arguments ${ARGN})
```

Comment:
this looks specious.

You are using a macro, meaning same scoping applies inside as outside the macro.

afaict there is not specific need for same scoping in this case, so I would like to know why a function is not used in this case.